run.py

```
@@ -227,24 +290,121 @@ async def get_historic_payments() -> List[Dict[str, Any]]:
+            return []
+
         if historic_df is not None and not historic_df.empty:
             records = historic_df.to_dict('records')
             logger.info(f"✅ Loaded {len(records)} historic payment records")
             return records
         else:
             logger.warning("No historic payment data available")
             return []
-            
+
     except Exception as e:
```

Ruff (0.13.3)

Do not catch blind exception: `Exception`

(BLE001)